### PR TITLE
InstructorFeedbackEditPageUiTest line 155 commented out #3818

### LINKED ISSUE #3818: InstructorFeedbackEditPageUiTest line 155 commented out
It's currently commented out without any comments explaining why, do we still need this test at all or has it been accidentally commented out?

screen shot 2015-06-24 at 4 05 28 pm

### PR DESCRIPTION
**Pull request by junhaoyap**
Fixes #3818

### CODE REVIEW

**Comment by MrFool** [comment]
Can be reviewed now :)

**Comment by MrFool** [comment]
@JosephineKwa Help me take a look when you're free :)

**Comment by JosephineKwa** [comment]
Sure!

File: src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java [hunk 1]
```diff
@@ -150,9 +150,11 @@ private void testEditSessionAction() throws Exception {
...
 
         feedbackEditPage = getFeedbackEditPage();
-        feedbackEditPage.isElementSelected(Const.ParamsNames.FEEDBACK_SESSION_RESULTSVISIBLEBUTTON
-                                           + "_atvisible");
-        //feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackEditPublished.html");
+
+        // Ensuring that the settings did not default back to original values after manual publishing
+        assertFalse(feedbackEditPage.isElementSelected(Const.ParamsNames.FEEDBACK_SESSION_RESULTSVISIBLEBUTTON
+                                                       + "_atvisible"));
+
```
Comment (JosephineKwa) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Just a question, why do we have to test if the publish immediately button isn't selected though? What does the 'original values' here refer to?

Comment (MrFool) [inline src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java]: Just below this (I don't think it's being shown in the diff), there is this "restore defaults" that clicks back on the so called "default button" I believe that is the immediate visible button that is being asserted to not be selected right here. Originally, the session was immediately visible but was edited from above to be something else.

Right above here, there is this verifyHtmlMainContent after the session has been edited. I believe the original intention of the test was to ensure that the "visibility option" stays as the edited option.

However, previously this test was doing feedbackEditPage.isElementSelected() which returns a boolean but does no assertion so I added the assertion.

**Comment by JosephineKwa** [comment]
Ah I see what you mean :) I tried stepping through the code, it seems that the original button that was selected above was the "at" button though, i.e. the custom publish button. Perhaps the person who wrote the test mistook `Const.ParamsNames.FEEDBACK_SESSION_RESULTSVISIBLEBUTTON + "_atvisible"` as the "at" button when it actually refers to the "immediately" button

**Comment by MrFool** [comment]
Yeah the test is not very concrete, i.e. that I'm also not very very sure what it's supposed to be testing for. Previously, about a year ago when this was removed, the only difference between the previous HTML verification and the one that was removed was a "status bar", i.e. that one of them had a status bar and one of them did not, they didn't differ anywhere else in HTML.

I kinda thought that the test was meant to ensure that the radio button being selected hasn't changed but since before I added the assertion there wasn't any assertion, I can only guess so I am looking for inputs to see whether this would be an appropriate test.

Perhaps we should test other radio buttons? Or perhaps we should still revert to a HTML check to make sure everything else remains the same after a manual publish?

**Comment by damithc** [comment]
It is possible that this test was added to catch a specific bug (to comply with the rule: write a test before fixing a bug). See if you can trace it back to the PR.

**Comment by junhaoyap** [comment]
Ok I'll do that first thing in the morning tomorrow.

**Comment by junhaoyap** [comment]
Looking at the PR, this is part of refactoring ->

screen shot 2015-06-29 at 11 14 54 am

However, after more thinking about this issue, I believe we should revert back to a HTML test because even after adding an assertion, the assertion does not assert enough stuff.

I believe the original intention of the test before it was changed to assert nothing in particular (note that the testing method .isElementSelected() returns a boolean but it wasn't asserted) was to ensure that after a manual publish, nothing in the page changes.

screen shot 2015-06-29 at 11 14 45 am

@damithc should I revert it to a full HTML verification or a main content HTML check?

**Comment by junhaoyap** [comment]
Refer to #1953

**Comment by damithc** [comment]
Yes, i don't mind a full html check.

**Comment by junhaoyap** [comment]
@damithc upgraded to a full html test

@JosephineKwa can please help to review? :)

**Comment by JosephineKwa** [comment]
Looks good to merge!

**Comment by junhaoyap** [comment]
Thanks for the review :+1:

**Pull request merged**